lib/file-list.js

```diff
@@ -95,13 +95,12 @@ class FileList {
               return this._preprocess(file).then(() => file)
```

Comment:
The `map()` operates on a sync array, returning either a sync array or an array of promises.  I guess sync `Array.map()` works fine for that somewhat funky operation.  It would be less funky if we `return Promise.resolve(file)` in the ` nocache` branch.